bgpd/bgp_route.c

```diff
@@ -9656,6 +9658,27 @@ void route_vty_out_detail(struct vty *vty, struct bgp *bgp, struct bgp_dest *bn,
 		}
 	}
 
+	if (safi == SAFI_EVPN
+	    && attr->evpn_overlay.type == OVERLAY_INDEX_GATEWAY_IP) {
+		char gwip_buf[INET6_ADDRSTRLEN];
+
+		if (is_evpn_prefix_ipaddr_v4((struct prefix_evpn *)&bn->p))
+			inet_ntop(AF_INET, &attr->evpn_overlay.gw_ip.ipv4,
+				  gwip_buf, sizeof(gwip_buf));
+		else
+			inet_ntop(AF_INET6, &attr->evpn_overlay.gw_ip.ipv6,
+				  gwip_buf, sizeof(gwip_buf));
+
+		if (json_paths)
+			json_object_string_add(json_path, "gatewayIP",
+					       gwip_buf);
+		else
+			vty_out(vty, " Gateway IP %s", gwip_buf);
```

Comment:
doesn't %pIA work?

Reply:
Will update

Reply:
%pI4 and %pI6 would work. But can we use it for json? In that case we will still need this buf for json.